xbmc/cores/AudioEngine/Encoders/AEEncoderFFmpeg.cpp

```diff
@@ -290,34 +290,33 @@ int CAEEncoderFFmpeg::Encode(uint8_t *in, int in_size, uint8_t *out, int out_siz
 
     avcodec_fill_audio_frame(frame, channelNum, m_CodecCtx->sample_fmt, in, in_size, 0);
 
-    pkt->size = out_size;
-    pkt->data = out;
-
     /* encode it */
     err = avcodec_send_frame(m_CodecCtx, frame);
     if (err < 0)
       throw FFMpegException("Error sending a frame for encoding (error '{}')",
                             FFMpegErrorToString(err));
 
-    while (err >= 0)
+    bool got_package = false;
+    while (!got_package)
     {
```

Comment:
in this comination the while loop makes no sense, as it can always only go through this code exactly once:

1) proper package
2) EAGAIN or EOF
3) Error

so in this combination you can change the while to an if instead or consider my second try in the original thread, which was tested.

Reply:
I simplified the commit and ensured there are no packet leaks thanks to early returns. Please test it again. @smf007 I will schedule an Android 32 and 64 build for you.